crates/sui-framework/packages/move-stdlib/sources/option.move

```diff
@@ -174,7 +171,7 @@ module std::option {
 
     /// Select the first `Some` value from the two options, or `None` if both are `None`.
     /// Equivalent to Rust's `a.or(b)`.
-    public macro fun or<$T>($o: Option<$T>, $default: Option<$T>): Option<$T> {
+    public macro fun or<$T: drop>($o: Option<$T>, $default: Option<$T>): Option<$T> {
```

Comment:
Impossible to pass a default argument without drop.

Reply:
It's a macro, it won't be evaluated otherwise. So this should be

```
public macro fun or<$T: drop>($o: Option<$T>, $default: Option<$T>): Option<$T> {
    let o = $o;
    if (o.is_some()) o
    else { o.destroy_none(); $default }
}
```

(modulo formatting)